src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4445,14 +4447,17 @@ private BoundExpression BindUnexpectedComplexElementInitializer(InitializerExpre
                     return boundCall;
                 }
 
+                bool invokedAsExtensionMethod = boundCall.InvokedAsExtensionMethod;
                 return new BoundCollectionElementInitializer(
                     elementInitializer,
                     boundCall.Method,
                     boundCall.Arguments,
+                    invokedAsExtensionMethod ? null : implicitReceiver,
```

Comment:
>invokedAsExtensionMethod ? null : implicitReceiver [](start = 20, length = 50)

It feels like rather than having this logic we should simply pull receiver from the boundCall, it should already be properly adjusted